src/snowflake/snowpark/session.py

```diff
@@ -805,6 +821,10 @@ def _resolve_imports(
     ) -> List[str]:
         """Resolve the imports and upload local files (if any) to the stage."""
         resolved_stage_files = []
+
+        if _is_execution_environment_sandboxed:
```

Comment:
nit: I'd add a comment here to explain what we're doing with the short-circuit

Reply:
Done, in the next commit.